distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFtp.java

```diff
@@ -0,0 +1,267 @@
+    public void testInsecureFtpPutStream() throws Exception {
+        client = createInsecureClient();
+
+        StringWriter writer = new StringWriter();
+        IOUtils.copy(getClass().getResourceAsStream("/metacard1.xml"), writer);
+        ftpPutStreaming(writer.toString());
+
+        // verify FTP PUT resulted in ingest, catalogued data
+        Response response = executeOpenSearch("xml", "q=*", "count=100");
+        response.then().log().all().body("metacards.metacard.size()", equalTo(1));
+
+        // clean up test data
+        TestCatalog.deleteMetacard(getMetacardIdFromResponse(response));
+    }
+
+    private FTPClient createInsecureClient() throws Exception {
+        FTPClient ftp = new FTPClient();
+
+        ftp.connect(FTP_SERVER, FTP_DEFAULT_PORT);
+        showServerReply(ftp);
+        int connectionReply = ftp.getReplyCode();
+        if (!FTPReply.isPositiveCompletion(connectionReply)) {
+            fail("FTP server refused connection: " + connectionReply);
+        }
+
+        boolean success = ftp.login(USERNAME, PASSWORD);
+        showServerReply(ftp);
+        if (!success) {
+            fail("Could not log in to the FTP server.");
+        }
+
+        ftp.enterLocalPassiveMode();
+        ftp.setControlKeepAliveTimeout(300);
+        ftp.setFileType(FTP.BINARY_FILE_TYPE);
+
+        return ftp;
+    }
+
+    private void disconnect() {
+        if (client != null && client.isConnected()) {
+            try {
+                client.logout();
+            } catch (IOException ioe) {
+                // ignore
+            }
+            try {
+                client.disconnect();
+            } catch (IOException ioe) {
+                // ignore
+            }
+        }
+    }
+
+    private Response executeOpenSearch(String format, String... query) {
+        StringBuilder buffer = new StringBuilder(OPENSEARCH_PATH.getUrl()).append("?")
+                .append("format=").append(format);
+
+        for (String term : query) {
+            buffer.append("&").append(term);
+        }
+
+        String url = buffer.toString();
+        LOGGER.info("Getting response to {}", url);
+
+        return when().get(url);
+    }
+
+    private void ftpPut(String data) throws IOException {
+        InputStream ios = new ByteArrayInputStream(data.getBytes());
+
+        LOGGER.info("Start data upload via FTP PUT...");
+        // file will not actually be written to disk on ftp server
+        boolean done = client.storeFile("test", ios);
+
+        ios.close();
+        showServerReply(client);
+
+        if (done) {
+            LOGGER.debug("File uploaded successfully.");
+        } else {
+            LOGGER.error("Failed to upload file.");
+        }
+    }
+
+    private void ftpPutStreaming(String data) throws IOException {
+        InputStream is = new ByteArrayInputStream(data.getBytes());
+
+        LOGGER.info("Start data upload via FTP PUT...");
+        OutputStream os = client.storeFileStream("test");
```

Comment:
A nit, but the try with resources could be used here for both InputStream and OutputStream